### DISCUSSION TITLE
Edge functions not found locally

### DISCUSSION THREAD

**Discussion by fsa317** [vercel/community#1698] [Help] [answered]
### Summary

Hi, I am playing with nextjs via Vercel.  I started with a simple page and edge function (located in pages/api/generate.ts).  It was working fine.  I then wanted to add a python serverless function.  After some messing around I got that working.  However, at some point my edge functions stopped working and now returns a 404 when I run locally (using vercel dev).

In my vercel.json I have the following ( to get the python function working):

```
{
    "rewrites": [
      { "source": "/pyapi/(.*)", "destination": "/api/index" }
    ],
    "cleanUrls": true
  }
```
The rewrite seemed to be what helped me get the Python serverless function working, but my edge functions are returning a 404 (locally, but works on the Vercel deployment).

My workflow is just running `vercel dev`, do I need to be running `vercel build`?  

Any help is appreciated.

Update:
The issue seems to be related to having a setup where I have:

/api/index.py
/pages/api/test.ts

That plus the rewrite above causes the issue where the test.ts is not being found.  If I move test.ts to /api/test.ts it works, but I get the following warning:

``` WARN! When using Next.js, it is recommended to place Node.js Serverless Functions inside of the `pages/api` (provided by Next.js) directory instead of `api` (provided by Vercel).```

Should having both `/api` and `/pages/api` work?

**Comment by amyegan** [comment]
Hi @fsa317. Do you have a public repo or a minimal reproduction that I can use to get a better look at the problem?

**Reply by fsa317** [reply]
I will create a simple reproduction of the issue soon. It will help me verify this wasn’t just me messing something up.

**Reply by fsa317** [reply]
Here is a reproduction.

https://github.com/fsa317/verceltest

If you run this via `vercel dev` locally the url `/pyapi/pytest` works but `/api/test` does not (yet it did before I added the python function).

Thanks again @amyegan

**Reply by fsa317** [reply]
Any updates on this?

**Reply by amyegan** [reply]
Sorry for the delay. I was out of office at the end of last week. I'm getting inconsistent results when testing, so I'm still trying to figure this one out. I'll let you know when I have more information

**Reply by amyegan** [reply, marked as answer]
After a bit of research, I found that using both `/api` and `/pages/api` can lead to unexpected behavior. I responded to your more recent post in more detail https://github.com/vercel/community/discussions/1998#discussioncomment-5546569
